homeassistant/components/homekit_controller/media_player.py

```diff
@@ -63,8 +65,15 @@ def get_characteristic_types(self):
             CharacteristicsTypes.CURRENT_MEDIA_STATE,
             CharacteristicsTypes.TARGET_MEDIA_STATE,
             CharacteristicsTypes.REMOTE_KEY,
+            CharacteristicsTypes.ACTIVE_IDENTIFIER,
+            # Characterics that are on the linked INPUT_SOURCE services
+            CharacteristicsTypes.CONFIGURED_NAME,
+            CharacteristicsTypes.IDENTIFIER,
         ]
 
+    def _setup_active_identifier(self, char):
+        self._features |= SUPPORT_SELECT_SOURCE
```

Comment:
Will this ever change during entity lifetime?

Reply:
Right now, no. Some TV's might support it some might not, but it will be determined when the TV is added to HA and not again (until restart). Characteristics don't come and go very frequently. If it changed it would probably be a firmware update that added more capabilities to a device that already had very basic HomeKit support in its firmware.

That said, in a future PR i plan to make it some that the setup phase /can/ be run again, idempotently. HomeKit accessories publish a change to their zeroconf metadata if what i call the entity map changes. I handle this already to some extent and add new entities that have been picked added to bridges. But I want to also reconfigure all entities in that case to pick up new features or altered limits.

Reply:
Ok. Supported features shouldn't change during entity lifetime. It could be ok to remove an old entity and add a new one in the old ones place if we want to reconfigure in the future.

Reply:
Ah thanks for the heads up - i'll bear that in mind!